### ISSUE TITLE
[Bug] Ads displayed in v18.08.1-2.3.1 using iPhone 13 Pro Max

### ISSUE THREAD

**Issue by eutudb23** [qnblackcat/uYouPlus#996]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Have you read the FAQ?

- [X] Yes, I did read the FAQ

### Device info

- iOS/iPadOS version: 16.3.1
- Device model: iPhone 13 Pro Max
- Sideload tool (AltStore, Sideloadly, TrollStore,...): AltStore
- The specific version of uYouPlus (**latest or newest is NOT a version number!**): v18.08.1-2.3.1

### Describe the bug

Ads are displayed before videos in the version specified.
The same version seems to work (not display ads) on iPad Pro M2 using iPadOs 16.13.1 .

### Steps to reproduce the issue

1. Install the version specified using AltStore
2. Open any youtube video
3. Ads are displayed before videos

### Are you using the newest version of uYouPlus? If not, why?

✅ Yes, I'm using the latest version of uYouPlus right now

### Does the issue happen with the official YouTube from AppStore?

❌ No, I can't reproduce this issue in the official YouTube from AppStore

**Comment by JackPlaysGames0510** [comment]
Go to the UYou setting and toggle it on

**Comment by eutudb23** [comment]
I saw, did that but the same expected outcome. Ads still come through.
At the moment I switched back to the January build.
Will test out when a new version comes out.

**Comment by eutudb23** [comment]
I found the issue, the setting is not persisted restarting the app, after setting from uYou, Other Settings.
Other Settings restores to default values after restarting uYou app.

**Comment by qnblackcat** [comment]
@eutudb23 @azalty read the FAQ and you'll know why those option are being disabled

**Comment by qnblackcat** [comment]
@eutudb23 I can't reproduce this issue. Not sure what kind of ads are you talking about?

**Comment by Carrotcruncheruk** [comment]
No ads here either, maybe a settings issue? Op should check all their settings it's easy to miss something.

**Comment by eutudb23** [comment]
> @eutudb23 @azalty read the FAQ and you'll know why those option are being disabled

Hi, yes I saw it mentioned in the FAQ. Due to timestamp is being default to off every time.
Either way, I would rather have the option permanently enabled (disable ads) than timestamps.
Worked great in previous release.
Thank for the great work, regardless of the issue above.

**Comment by qnblackcat** [comment]
@eutudb23 again, what kind of ads is that? Like normal ads?

**Comment by eutudb23** [comment]
> @eutudb23 again, what kind of ads is that? Like normal ads?

Yes, since the toggle is back to default turned off if restarting the app.

**Comment by qnblackcat** [comment]
@eutudb23 screenshots?

**Comment by eutudb23** [comment]
> @eutudb23 screenshot of the ads?

After updating to iOS 16.4 seems to be working now and what is funny the settings are not enabled 😄))) will see after next refresh or re-install.
Will update then if still the same or close this issue.

**Comment by leewhat** [comment]
I'm also getting ads these few days, iOS 14.8.1 on iPhone 6s+.
Ads on top of home screen, ads before video playback, ads inbetween couple Shorts.

**Comment by androo-d** [comment]
I have been having the same issue over the past week. The toggled are enabled, I check after every ad I get. It seems like most ads are being blocked but sometimes I get a few.

**Comment by zoplo** [comment]
Having same issue, get ads sometimes before videos starts

**Comment by zoplo** [comment]
![Uploading DDC095AD-71B5-42F6-9ABE-BA6A86DDB436.png…]()

**Comment by zoplo** [comment]
![DDCFF101-C920-4271-801E-58C43D4ACDB7](https://user-images.githubusercontent.com/66632254/229247205-aad82348-4aed-4f60-9463-59e09cb9cd4c.png)

**Comment by zoplo** [comment]
> > @eutudb23 screenshot of the ads?
> 
> After updating to iOS 16.4 seems to be working now and what is funny the settings are not enabled 😄))) will see after next refresh or re-install. Will update then if still the same or close this issue.

Did updating to iOS 16.4 fix the issue for you?

**Comment by eutudb23** [comment]
> > > @eutudb23 screenshot of the ads?
> 
> > 
> 
> > After updating to iOS 16.4 seems to be working now and what is funny the settings are not enabled 😄))) will see after next refresh or re-install. Will update then if still the same or close this issue.
> 
> 
> 
> Did updating to iOS 16.4 fix the issue for you?

Yes, which is wired and makes no sense to me 😄. As mentioned above I don't even have the setting toggled to On and get no ads after I updated to 16.4

**Comment by zoplo** [comment]
> > > > @eutudb23 screenshot of the ads?
> > 
> > 
> > > 
> > 
> > 
> > > After updating to iOS 16.4 seems to be working now and what is funny the settings are not enabled 😄))) will see after next refresh or re-install. Will update then if still the same or close this issue.
> > 
> > 
> > Did updating to iOS 16.4 fix the issue for you?
> 
> Yes, which is wired and makes no sense to me 😄. As mentioned above I don't even have the setting toggled to On and get no ads after I updated to 16.4. Updated my wife's iPhone 11 Pro Max to iOS 16.4 and the same, Ads setting off and No ads.

Hmm weird, just updated to iOS 16.4 and still getting ads. Did you have to reinstall uYou+?

**Comment by eutudb23** [comment]
> > > > > @eutudb23 screenshot of the ads?
> 
> > > 
> 
> > > 
> 
> > > > 
> 
> > > 
> 
> > > 
> 
> > > > After updating to iOS 16.4 seems to be working now and what is funny the settings are not enabled 😄))) will see after next refresh or re-install. Will update then if still the same or close this issue.
> 
> > > 
> 
> > > 
> 
> > > Did updating to iOS 16.4 fix the issue for you?
> 
> > 
> 
> > Yes, which is wired and makes no sense to me 😄. As mentioned above I don't even have the setting toggled to On and get no ads after I updated to 16.4. Updated my wife's iPhone 11 Pro Max to iOS 16.4 and the same, Ads setting off and No ads.
> 
> 
> 
> Hmm weird, just updated to iOS 16.4 and still getting ads. Did you have to reinstall uYou+?

On one device yes, on the other one just to refresh.

**Comment by leewhat** [comment]
seems working now after removing uYou+ then reinstall.

**Comment by zoplo** [comment]
> seems working now after removing uYou+ then reinstall.

The latest version?

**Comment by leewhat** [comment]
> > seems working now after removing uYou+ then reinstall.
> 
> The latest version?

the latest but the ads returning again after a while :(

**Comment by eutudb23** [comment]
@qnblackcat just re-installed latest version again as it expired and seems to display Ads with the option enabled.
Getting ads before video start and in between.

![image](https://user-images.githubusercontent.com/11492145/229300053-24f01e4c-d583-4593-a4bd-e8d19c00c1c7.png)
![image](https://user-images.githubusercontent.com/11492145/229300084-75050761-fa34-4ad3-b9e8-4f6091352e5b.png)

**Comment by leewhat** [comment]
just making sure, uYou is v2.1? 
![IMG_5925](https://user-images.githubusercontent.com/1757959/229303388-642694ef-ac89-4fb5-bb8c-9cbbea94d8d4.JPG)

**Comment by eutudb23** [comment]
> just making sure, uYou is v2.1? 
> 
> ![IMG_5925](https://user-images.githubusercontent.com/1757959/229303388-642694ef-ac89-4fb5-bb8c-9cbbea94d8d4.JPG)
> 
> 

Yes, check the reported issue. It is specified the build version.

**Comment by level3tjg** [comment]
> seems working now after removing uYou+ then reinstall.
> 
> edit: nvm, spoke too soon.

If it's only working shortly after reinstall then it's likely being caused by an A/B test or some server-sided change, that also means it's very unlikely we'll be able to reproduce the same behavior so it's harder to fix. Does adblock still not work when you're in incognito mode?

**Comment by leewhat** [comment]
> If it's only working shortly after reinstall then it's likely being caused by an A/B test or some server-sided change, that also means it's very unlikely we'll be able to reproduce the same behavior so it's harder to fix. Does adblock still not work when you're in incognito mode?

doesn't work as well in incognito.

**Comment by qnblackcat** [comment]
https://github.com/qnblackcat/uYouPlus/suites/11980500434/artifacts/628707815

Try this version? It’s a zip file so you have to extract it first

**Comment by leewhat** [comment]
> https://github.com/qnblackcat/uYouPlus/suites/11980500434/artifacts/628707815
> 
> Try this version? It’s a zip file so you have to extract it first

so far no ads on homescreen and video but still have ads between Shorts.

**Comment by qnblackcat** [comment]
@level3tjg could you take a look at this?

**Comment by level3tjg** [comment]
I think it's caused by `-[YTHotConfig disableAfmaIdfaCollection]` being enabled. Temporary solution is to disable the flag in YTABC, in code we'll need to hook `+[YTDataUtils spamSignalsDictionaryWithoutIDFA]` and return nil

**Comment by leewhat** [comment]
ads in shorts seems gone after disable that, thank you.

**Comment by androo-d** [comment]
> I think it's caused by `-[YTHotConfig disableAfmaIdfaCollection]` being enabled. Temporary solution is to disable the flag in YTABC, in code we'll need to hook `+[YTDataUtils spamSignalsDictionaryWithoutIDFA]` and return nil

This has done the trick for me. Thank you

**Comment by zoplo** [comment]
Can I ask where I disable that option? Can’t seem to find it

**Comment by androo-d** [comment]
> Can I ask where I disable that option? Can’t seem to find it

In the A/B options under normal YouTube settings. Enable A/B then turn off disableAfmaIdfaCollection.

**Comment by zoplo** [comment]
> > Can I ask where I disable that option? Can’t seem to find it
> 
> In the A/B options under normal YouTube settings. Enable A/B then turn off disableAfmaIdfaCollection.

Perfect thank you
